### DISCUSSION TITLE
slow response

### DISCUSSION THREAD

**Discussion by rajdeepbanerjee-git** [Holmeswww/AgentKit#7] [Q&A] [answered]
Hi, 

Thank you for the nice work. I used the example code given with my API key, the graph.evaluate() keeps running for more than 5 mins even for a simple task. How to check where I am going wrong?

`graph = Graph()

subtask1 = "Consider that Ram can eat 20 apples in a day."
node1 = BaseNode(subtask1, subtask1, graph, LLM_API_FUNCTION, agentkit.compose_prompt.BaseComposePrompt())
graph.add_node(node1)

subtask2 = "Consider that Ram eat 1 apple per day in general."
node2 = BaseNode(subtask2, subtask2, graph, LLM_API_FUNCTION, agentkit.compose_prompt.BaseComposePrompt())
graph.add_node(node2)

subtask3 = "Now, tell me how many apples Ram can eat in 2 days and how many he will generally eat in 2 days?"
node3 = BaseNode(subtask3, subtask3, graph, LLM_API_FUNCTION, agentkit.compose_prompt.BaseComposePrompt())
graph.add_node(node3)

graph.add_edge(subtask1, subtask2)
graph.add_edge(subtask1, subtask3)
graph.add_edge(subtask2, subtask3)

result = graph.evaluate()`

**Comment by Holmeswww** [comment, marked as answer]
Hello,

Thanks for bringing this issue to our attention. This is most likely an OpenAI issue.

Our LLM_API tries to catch OpenAI errors for robust multi-turn agent performance, and it might be catching too many errors.

df43b1e

Please install the latest version v0.1.5 for the changes to take effect

We have updated LLM_API to expose **OpenAI authentication errors**.

In addition, we updated starter code to use GPT-4-turbo for a faster experience.

Finally, when setting the nodes, consider passing the `verbose=True` flag, this will show the progress within the node. See our example below:


```
wandb Trace seems to be missing. This is fine if you are not using wandb
Environment variable for OpenAI key not found, using OpenAI API key from ~/.openai/openai.key.
Matched model name to:  ['gpt-4-turbo', 'gpt-3.5-turbo', 'gpt-3.5-turbo-1106']
Prompt: Consider that Ram can eat 20 apples in a day.
Answer: That's quite a lot of apples for one person to eat in a day! If Ram eats 20 apples each day, he would consume 140 apples in a week, and about 600 apples in a month. This is assuming he maintains the same rate of consumption every day. Eating such a large number of apples regularly could have various health implications, both positive and negative, depending on his overall diet and health condition.

Prompt: Consider that Ram eat 1 apple per day in general.
Answer: If Ram eats 1 apple per day, that's a more typical amount. Eating an apple a day is often considered healthy due to apples being rich in important nutrients, fiber, and antioxidants. This habit can contribute to a balanced diet and may offer health benefits such as improving digestion, reducing the risk of chronic diseases, and aiding in weight management.

Over the course of a week, Ram would eat 7 apples, and in a month, he would consume about 30 apples, assuming each month has around 30 days. This moderate intake makes it easier to maintain a varied diet while still enjoying the health benefits of apples.

Prompt: Now, tell me how many apples Ram can eat in 2 days and how many he will generally eat in 2 days?
Answer: Based on the scenarios you've provided:

1. **If Ram can eat 20 apples in a day**, then in 2 days, he can eat \(20 \times 2 = 40\) apples.

2. **If Ram generally eats 1 apple per day**, then in 2 days, he will generally eat \(1 \times 2 = 2\) apples.
```

**Reply by rajdeepbanerjee-git** [reply]
Hi, Thank you for the suggestions. I tried with your suggestion with the latest git version agentkit-llm-0.1.5. It starts to prompt but is taking a lot of time without showing error even with verbose =  True at the node definition. See below screenshot:
<img width="626" alt="agentkit_issue" src="https://github.com/Holmeswww/AgentKit/assets/12594346/aeaa5a2e-2604-49d4-a728-b335a7b39e82">

Appreciate any help to resolve this. Thanks!

**Reply by Holmeswww** [reply]
Hi,

The code might have been stuck in the LLM_API somewhere. What does it say when you terminate the code? (with keyboardInterrupt)

**Reply by rajdeepbanerjee-git** [reply]
You are right - it seems to be an authentication error with the incorrect org name! But I checked, the name is correct. Should I pass the organization id instead?

**Reply by Holmeswww** [reply]
Yes, you should use Org ID.

**Reply by rajdeepbanerjee-git** [reply]
Yup, now it works - thanks a lot! Sorry for bothering you with such a trivial issue! Thanks again for the beautiful work!
